orsopy/fileio/orso.py

```diff
@@ -273,3 +276,111 @@ def load_orso(fname: Union[TextIO, str]) -> List[OrsoDataset]:
         od = OrsoDataset(o, data)
         ods.append(od)
     return ods
+
+
+def _from_nexus_group(group):
+    if group.attrs.get("sequence", None) is not None:
+        sort_list = [[v.attrs["sequence_index"], v] for v in group.values()]
+        return [_get_nexus_item(v) for _, v in sorted(sort_list)]
+    else:
+        dct = dict()
+        for name, value in group.items():
+            if value.attrs.get("NX_class", None) == "NXdata":
+                # remove NXdata folder, which exists only for NeXus plotting
+                continue
+            dct[name] = _get_nexus_item(value)
+
+        ORSO_class = group.attrs.get("ORSO_class", None)
+        if ORSO_class is not None:
+            if ORSO_class == "OrsoDataset":
+                # TODO: remove swapaxes if order of data is changed (PR #107)
+                # reorder columns so column index is second:
+                dct["data"] = np.asarray(dct["data"]).swapaxes(0, 1)
+            cls = ORSO_DATACLASSES[ORSO_class]
+            return cls(**dct)
+        else:
+            return dct
+
+
+def _get_nexus_item(value):
+    import json
+
+    import h5py
+
+    if isinstance(value, h5py.Group):
+        return _from_nexus_group(value)
+    elif isinstance(value, h5py.Dataset):
+        v = value[()]
+        if isinstance(v, h5py.Empty):
+            return None
+        elif value.attrs.get("mimetype", None) == JSON_MIMETYPE:
+            return json.loads(v)
+        elif hasattr(v, "decode"):
+            # it is a bytes object, should be string
+            return v.decode()
+        else:
+            return v
+
+
+def load_nexus(fname: Union[str, BinaryIO]) -> List[OrsoDataset]:
```

Comment:
I think it would be advantageous to have a single load_orso function that chooses which loader to use automatically. That would allow external software to be agnostic about the file type that's used.
We can do this in the next release, though.

Suggested way of implementation:
1. load_orso scans through a list of known data format classes
2. Each format class has a static method to check if it supports the file in questoin (e.g. first header line for ort, hdf header for .orb)
3. If it's supported, the function uses the format class to retrieve the Orso objects from the file
4. Potentially we could store the format used to unpack in the Orso class